src/datastore/pg-write-store.ts

```diff
@@ -759,6 +767,34 @@ export class PgWriteStore extends PgStore {
     }
   }
 
+  // Find any transactions that are erroneously still marked as both `pending` in the mempool table
+  // and also confirmed in the mined txs table. Mark these as pruned in the mempool and log warning.
+  // This must be called _after_ any writes to txs/mempool tables during block and microblock ingestion,
+  // but _before_ any reads or view refreshes that depend on the mempool table.
+  // NOTE: this is essentially a work-around for whatever bug is causing the underlying problem.
+  async reconcileMempoolStatus(sql: PgSqlClient): Promise<void> {
```

Comment:
While this works great, I think we should add another similar check inside `updateMempoolTxs` that inserts them as `pruned` initially if we already have that same incoming mempool tx in a block (in a scenario where we receive a mempool tx with some huge delay for any reason).

That way, we won't display the inconsistent result from the time that delayed mempool tx arrives until the next block/microblock comes and we reconcile here.

What do you think?